src/oc_telemetry.erl

```diff
@@ -42,38 +23,57 @@ attach(Name, EventName, Description, Unit) ->
 track(#{'__struct__' := Type,
         name := NormalizedName,
         event_name := EventName,
-        metadata := Metadata,
+        measurement := Measurement,
         tags := Tags,
-        description := Desc,
+        tag_values := TagValues,
+        description := Description,
         unit := Unit} = Data) ->
   Name = build_name(NormalizedName),
-  Description = Desc,
-
-  case attach(Name, EventName, Description, Unit, Metadata) of
+  case attach(Name, EventName, Measurement, Description, Unit, TagValues) of
     {ok, Measure} ->
       Aggregation = aggregation(Type, Data),
-
       oc_stat_view:subscribe(Name, Measure, Description, Tags, Aggregation);
     Error -> Error
   end.
 
 % ------------------------------------------------------------------------------
 
-attach(Name, EventName, Description, Unit, Metadata) ->
-    Measure = oc_stat_measure:new(Name, Description, Unit),
-    Options = #options{name = Name, filter = Metadata},
-
-    case telemetry:attach(Name, EventName, fun handle_event/4, Options) of
-      ok -> {ok, Measure};
-      Error -> Error
+attach(Name, EventName, Measurement, Description, Unit, TagValues) ->
```

Comment:
I would add `attach/5` which would set `TagValues` to identity function by default. IMHO it would provide nicer API for most common use-case.

Reply:
Oh right, wanted to ask about that. I don't get why there are exported attach functions? Wouldn't this only be used for `track`?

Reply:
The idea was to support "manually" adding new listeners without `Telemetry.Metrics`.

Reply:
Ah, duh, ok.